lib/Environment.php

```diff
@@ -485,4 +487,13 @@ public static function slugify(string $text) : string
 
         return $text;
     }
+
+    private function addMissingReferenceSectionError(string $section) : void
+    {
+        $this->errorManager->error(sprintf(
+            'Unknown reference section "%s"%s',
+            $section,
+            $this->getCurrentFileName() !== '' ? sprintf(' in "%s" ', $this->getCurrentFileName()) : ''
```

Comment:
I think we have this code in a few places now. Might be time to create a shared helper method? We can just log an issue for that to do in another PR.